src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,520 @@
+   * This is a first-generation FSx for ONTAP file system.
+   */
+  MULTI_AZ_1 = 'MULTI_AZ_1',
+
+  /**
+   * A high availability file system configured for Multi-AZ redundancy to tolerate temporary AZ unavailability.
+   * This is a second-generation FSx for ONTAP file system.
+   */
+  MULTI_AZ_2 = 'MULTI_AZ_2',
+
+  /**
+   * A file system configured for Single-AZ redundancy.
+   * This is a first-generation FSx for ONTAP file system.
+   */
+  SINGLE_AZ_1 = 'SINGLE_AZ_1',
+
+  /**
+   * A file system configured with multiple high-availability (HA) pairs for Single-AZ redundancy.
+   * This is a second-generation FSx for ONTAP file system.
+   */
+  SINGLE_AZ_2 = 'SINGLE_AZ_2',
+}
+
+/**
+ * The configuration for the Amazon FSx for NeetApp ONTAP file system.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-fsx-filesystem-ontapconfiguration.html
+ */
+export interface OntapConfiguration {
+  /**
+   * The number of days to retain automatic backups.
+   * Setting this property to 0 disables automatic backups.
+   * You can retain automatic backups for a maximum of 90 days.
+   *
+   * @default - disable automatic backups
+   */
+  readonly automaticBackupRetention?: Duration;
+
+  /**
+   * Start time for 30-minute daily automatic backup window in Coordinated Universal Time (UTC).
+   *
+   * @default - no backup window
+   */
+  readonly dailyAutomaticBackupStartTime?: aws_fsx.DailyAutomaticBackupStartTime;
+
+  /**
+   * The FSx for ONTAP file system deployment type to use in creating the file system.
+   */
+  readonly deploymentType: OntapDeploymentType;
+
+  /**
+   * The total number of SSD IOPS provisioned for the file system.
+   *
+   * The minimum and maximum values for this property depend on the value of HAPairs and StorageCapacity.
+   * The minimum value is calculated as StorageCapacity * 3 * HAPairs (3 IOPS per GB of StorageCapacity).
+   * The maximum value is calculated as 200,000 * HAPairs.
+   *
+   * @default - 3 IOPS per GB of storage capacity
+   */
+  readonly diskIops?: number;
+
+  /**
+   * The IP address range in which the endpoints to access your file system will be created.
+   *
+   * You can have overlapping endpoint IP addresses for file systems deployed in the same VPC/route tables, as long as they don't overlap with any subnet.
+   *
+   * @default - an unused IP address range from the 198.19.* range
+   */
+  readonly endpointIpAddressRange?: string;
+
+  /**
+   * The ONTAP administrative password for the `fsxadmin` user with which you administer your file system using the NetApp ONTAP CLI and REST API.
+   *
+   * If you don't specify a password, Amazon FSx will not set one. In that case, the user will not be able to log in.
+   *
+   * You can change the admin password at any time through the management console.
+   *
+   * @default - do not set an admin password
+   */
+  readonly fsxAdminPassword?: string;
```

Comment:
Can we use the `SecretValue` type?

Reply:
I've updated my code to use `SecretValue`.

Reply:
I'm not very familiar with how to use SecretValue, so could you explain it to me?

Is it correct to assume that the typical use case is passing something created with `SecretValue.secretsManager()` as an argument? In that case, my understanding is that it would no longer be possible to validate the secret itself (such as checking the length), is that correct?

Reply:
> Is it correct to assume that the typical use case is passing something created with SecretValue.secretsManager() as an argument?

Yeah, users can specify the password with `SecretValue.secretsManager` (or `SecretValue.unsafePlainText`):

- `secretsManager`

```ts
const fsxProps = {
  // ...
  fsxAdminPassword: SecretValue.secretsManager('my-secrets-arn-or-id')
}
```

- `unsafePlainText` (that is raw string, so it's unsafe)

```ts
const fsxProps = {
  // ...
  fsxAdminPassword: SecretValue.unsafePlainText('MyUnsafeAdminPassword'),
}
```

We contributor use `unsafeUnwrap()` method to extract the value in constructs.

```ts
fsxAdminPassword = props.fsxAdminPassword?.unsafeUnwrap()
```

Then the string value is printed in CFn:

```
// secretsManager
"FsxAdminPassword": "{{resolve:secretsmanager:my-secrets-arn-or-id:SecretString:::}}",

// unsafePlainText
"FsxAdminPassword": "MyUnsafeAdminPassword"
```

> In that case, my understanding is that it would no longer be possible to validate the secret itself (such as checking the length), is that correct?

Yes, we have discussed password validation, but this would no longer be necessary.

https://github.com/open-constructs/aws-cdk-library/pull/42#discussion_r1780030824

Reply:
Thank you for your kindly explanation! I didn't know how secret is taken in CFn.